### ISSUE TITLE
vcpkg built with VC2017 fails with an exception

### ISSUE THREAD

**Issue by plbpietrz** [Microsoft/vcpkg#855]
Hi,

I have a clean, up to date Win10 Aniv install with VC2017. Building `vcpkg.exe` using `bootstrap.ps1` results in a executable that crashes with an exception when run from the command line without any args. Below is the stacktrace:
```
ucrtbase.dll!__crt_char_traits<wchar_t>::tcslen<wchar_t * &>(wchar_t * &)
ucrtbase.dll!<lambda>(void)()
ucrtbase.dll!construct_environment_block<wchar_t>()
ucrtbase.dll!common_pack_argv_and_envp<wchar_t>()
ucrtbase.dll!execute_command<wchar_t>()
ucrtbase.dll!common_spawnv<wchar_t>()
ucrtbase.dll!__wspawnve()
ucrtbase.dll!common_spawnvp<wchar_t>()
ucrtbase.dll!__wspawnvpe()
ucrtbase.dll!common_spawnlp<wchar_t>()
ucrtbase.dll!__wspawnlpe()
vcpkg.exe!vcpkg::System::cmd_execute_clean(const wchar_t * cmd_line) Line 87
vcpkg.exe!vcpkg::Flush() Line 421
[Inline Frame] vcpkg.exe!wmain::__l2::<lambda_f693fcbd3de73d4e9140ed341d57efb2>::operator()() Line 187
vcpkg.exe!<lambda_f693fcbd3de73d4e9140ed341d57efb2>::<lambda_invoker_cdecl>() Line 188
[External Code]	
vcpkg.exe!vcpkg::Checks::exit_with_code(const vcpkg::LineInfo & line_info, const int exit_code) Line 30
vcpkg.exe!vcpkg::Checks::exit_fail(const vcpkg::LineInfo & line_info) Line 14
vcpkg.exe!inner(const vcpkg::vcpkg_cmd_arguments & args) Line 36
vcpkg.exe!wmain(const int argc, const wchar_t * const * const argv) Line 219
```

**Comment by ras0219-msft** [comment]
After looking through the code at those locations, I'm unable to determine what the issue is.

Could you open up the `toolsrc\vcpkg.sln` solution, hit yes to the upgrade prompt, build a debug version, and try to repro the issue again? You can run the built vcpkg.exe directly from the IDE -- it'll figure out the right root folder.

**Comment by ChrisTX** [comment]
I got the same issue on my laptop, but not on my desktop computer, which is setup almost identically.

Tried debugging it to no avail:

- It crashes while trying to upload ApplicationInsights metrics. If you build with DISABLE_METRICS=1 it doesn't occur. Note that this means Debug builds won't crash by default, unless you change the metrics behavior in metrics.cpp at which point they'll also crash.

- It only crashes when launching from PowerShell or from Windows Explorer. It doesn't crash when launching from cmd or from VS.

- The crash always occurs in the construct_environment_block part.

After some research I found this [issue on the Jython tracker](https://bugs.jython.org/issue29908) and [this issue on the wpdev uservoice](https://wpdev.uservoice.com/forums/110705-universal-windows-platform/suggestions/18355615-ucrtbase-bug-wspawnve-is-broken). It's probably that, but from what I can tell, the environment passed to the function is identical - and it's strange that my laptop and desktop with the exact same Windows version, Visual Studio version and Windows SDK behave differently here.
